src/librustc/ty/mod.rs

```diff
@@ -755,6 +755,9 @@ pub enum Predicate<'tcx> {
     /// for some substitutions `...` and T being a closure type.
     /// Satisfied (or refuted) once we know the closure's kind.
     ClosureKind(DefId, ClosureKind),
+
+    /// `T1 <: T2`
+    Subtype(PolySubtypePredicate<'tcx>),
```

Comment:
Can subtype predicates ever be higher-ranked (aka `for<'a> &'a u32 <: &'b u32`) ? I don't think so after this patch. Are you trying to support that in the future?